### PR TITLE
AndroidX for Fabulous.XamarinForms.Templates

### PR DESCRIPTION
**Pull request by Happypig375**
The pull request I mentioned at https://github.com/dotnet/maui/issues/115#issuecomment-635163230.

### CODE REVIEW

File: Fabulous.XamarinForms/templates/content/blank/NewApp.Android/NewApp.Android.fsproj [hunk 1]
```diff
@@ -270,34 +240,114 @@
     <Reference Include="Fabulous.XamarinForms.LiveUpdate">
       <HintPath>..\packages\Fabulous.XamarinForms.LiveUpdate.FabulousPkgsVersion\lib\netstandard2.0\Fabulous.XamarinForms.LiveUpdate.dll</HintPath>
     </Reference>
-    <ProjectReference Include="..\NewApp\NewApp.fsproj">
-      <Project>{99E19497-29A6-4B77-B773-BEC55F9B55DC}</Project>
-      <Name>NewApp</Name>
+    <ProjectReference Include="..\Fabulousv0_54_1Template\Fabulousv0_54_1Template.fsproj">
+      <Project>{2C65EDFC-6BF7-4087-AEC1-B4B0CE145CB7}</Project>
+      <Name>Fabulousv0_54_1Template</Name>
```
Comment (TimLariviere) [inline Fabulous.XamarinForms/templates/content/blank/NewApp.Android/NewApp.Android.fsproj:245]: Unwanted changes slipped through 😄 
Could you revert that passage, please?

File: Fabulous.XamarinForms/templates/content/blank/.template.config/template.json [hunk 2]
```diff
@@ -120,31 +120,265 @@
...
-      "defaultValue": "1.1.1.3"
+      "replaces": "XamarinAndroidXAnnotationVersion",
+      "defaultValue": "1.1.0"
+    },
+    "XamarinAndroidXAppCompatVersion": {
+      "type": "parameter",
+      "datatype": "string",
+      "replaces": "XamarinAndroidXAppCompatVersion",
+      "defaultValue": "1.1.0"
+    },
```
Comment (TimLariviere) [inline Fabulous.XamarinForms/templates/content/blank/.template.config/template.json:154]: Is it possible to aggregate several packages under a single token when they share the same version?
Like `XamarinAndroidXVersion` => `1.1.0`, `XamainAndroidXArchCoreVersion` => `2.1.0`, etc.
Would be easier to maintain I think. There's a lot of packages in Android. 😄

File: Fabulous.XamarinForms/templates/content/blank/NewApp.Android/NewApp.Android.fsproj [hunk 3]
```diff
@@ -68,47 +83,11 @@
     <AndroidResource Include="Resources\values\colors.xml" />
     <None Include="packages.config" />
   </ItemGroup>
-  <Import Project="$(MSBuildExtensionsPath)\Xamarin\Android\Xamarin.Android.FSharp.targets" />
```
Comment (TimLariviere) [inline Fabulous.XamarinForms/templates/content/blank/NewApp.Android/NewApp.Android.fsproj:71]: This one is required. We should keep it.

**Comment by TimLariviere** [comment]
I guess the build issue is linked to the missing `Xamarin.Android.FSharp.targets`.

File: Fabulous.XamarinForms/templates/content/blank/.template.config/template.json (hunk 2 above)
Comment (Happypig375) [reply Fabulous.XamarinForms/templates/content/blank/.template.config/template.json:154]: Unlike in Android Support libraries, each AndroidX package can have an independent version. If we forcibly group packages together, we lose flexibility in updating.

File: Fabulous.XamarinForms/templates/content/blank/NewApp.Android/NewApp.Android.fsproj (hunk 3 above)
Comment (Happypig375) [reply Fabulous.XamarinForms/templates/content/blank/NewApp.Android/NewApp.Android.fsproj:71]: Readded

File: Fabulous.XamarinForms/templates/content/blank/.template.config/template.json (hunk 2 above)
Comment (Happypig375) [reply Fabulous.XamarinForms/templates/content/blank/.template.config/template.json:154]: You can see that there are `1.0.0`, `1.0.1`, `1.1.0`, `2.0.1` and `2.1.0`. An update of an individual package will not affect other packages.

**Comment by Happypig375** [comment]
I don't use Shell, and I haven't noticed anything wrong with my app yet.

**Comment by Happypig375** [comment]
Why is Mac build failing?

**Comment by TimLariviere** [comment]
From the build logs:
>This project references NuGet package(s) that are missing on this computer. Use NuGet Package Restore to download them.  For more information, see http://go.microsoft.com/fwlink/?LinkID=322105. The missing file is ../packages/Xamarin.AndroidX.MultiDex.2.0.1/build/monoandroid90/Xamarin.AndroidX.MultiDex.targets.

**Comment by Happypig375** [comment]
I can't figure out why this is failing. The package list should be correct.

**Comment by TimLariviere** [comment]
I missed that the import of NuGet.targets was removed. So Android weren't restoring its packages.
But now, it is complaining about `resource style/Theme.AppCompat.Light.Dialog (aka com.companyname:style/Theme.AppCompat.Light.Dialog) not found.`.

File: Fabulous.XamarinForms/templates/content/blank/.template.config/template.json (hunk 2 above)
Comment (TimLariviere) [reply Fabulous.XamarinForms/templates/content/blank/.template.config/template.json:154]: Ok

**Comment by Happypig375** [comment]
...

**Comment by TimLariviere** [comment]
Yeah, sorry for that. Azure DevOps is restrictive on that.

**Comment by Happypig375** [comment]
This commit should be it. Having that property set to true actually makes Visual Studio set TargetFramework to v9.0 and TargetAndroidSDK to 28 instead of 29.

**Comment by TimLariviere** [comment]
Hmm. Is it because MSBuild picks the wrong version of the Android Build tools?
```
_ResolveMonoAndroidSdks:
  MonoAndroid Tools: /Library/Frameworks/Xamarin.Android.framework/Libraries/xbuild/Xamarin/Android/
  Android Platform API level: 29
  TargetFrameworkVersion: v10.0
  Android NDK: /Users/runner/Library/Android/sdk/ndk-bundle/
  Android SDK: /Users/runner/Library/Android/sdk/
  Android SDK Build Tools: /Users/runner/Library/Android/sdk/build-tools/28.0.3/
  Java SDK: /Library/Java/JavaVirtualMachines/zulu-8.jdk/Contents/Home/
  Application Java class: android.app.Application
```

**Comment by Happypig375** [comment]
Should be. The template executes fine locally.

**Comment by TimLariviere** [comment]
It doesn't seem to be that. Let's try bumping Mono to the latest version.

**Comment by Happypig375** [comment]
@TimLariviere Can you verify that the template executes successfully locally? It does on my side.

**Comment by TimLariviere** [comment]
I have indeed the same issue locally. I'm investigating why.

**Comment by TimLariviere** [comment]
Ok. Seems like it was an issue with the NuGet restore process.
I removed the `.nuget` folder and manually ran `nuget restore` before building the templates.
NuGet.targets was an old way of restoring packages and I finally managed to get rid of it.

**Comment by Happypig375** [comment]
:tada:

**Pull request merged**